### PR TITLE
somacore 1.0.3

### PR DESCRIPTION
**Pull request by johnkerl**
In support of https://github.com/TileDB-Inc/tiledbsoma-feedstock/pull/26 for https://github.com/single-cell-data/TileDB-SOMA/issues/1455

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -23,12 +23,12 @@ requirements:
     - pip
   run:
     - python >=3.7
-    - anndata
-    - attrs >=22.1
+    - anndata >= 0.9.1
```
Comment (thetorpedodog) [inline recipe/meta.yaml]: Based on the build error, it looks like you need to not have a space between the operator and the number:

```
    - anndata >=0.9.1
```

(also pyarrow below)

**Review by thetorpedodog — COMMENTED** [review]
I don’t feel good enough about conda to approve this but it does look right to me, apart from:

**Review by nguyenv — APPROVED** [review]
I do not know if the versions for the run dependencies are correct or not, but everything else looks good.

**Comment by johnkerl** [comment]
@nguyenv I took them from `git diff -r python-v1.0.2 -r python-v1.0.3` for the non-Conda files in the repo https://github.com/single-cell-data/SOMA ...

**Comment by jdblischak** [comment]
For the version constraints of the run dependencies, shouldn't we follow what is defined in `pyproject.toml`?

https://github.com/single-cell-data/SOMA/blob/python-v1.0.3/pyproject.toml

Specifically:


```toml
dependencies = [
  "anndata",
  "attrs>=22.1",
  "numba",
  "numpy>=1.21",
  "pandas",
  "pyarrow",
  "scipy",
  "typing-extensions>=4.1", # For LiteralString
]
requires-python = "~=3.7"

module = ["anndata", "numba", "pandas", "pyarrow", "scipy"]
```

My understanding of the various frozen requirements files in `python-spec/` is that they are used for CI testing, and don't represent the minimum required versions

https://github.com/single-cell-data/SOMA/blob/main/python-spec/update-requirements-txt
https://github.com/single-cell-data/SOMA/blob/python-v1.0.3/.github/workflows/python-somacore.yaml#L27

**Comment by johnkerl** [comment]
Something _very_ weird: I did a local commit for @jdblischak 's excellent advice:

```
$ git log -n 4
commit b1cde6004425eea791e50ac1843b0f7d59421285 (HEAD -> kerl/somacore-1.0.3, origin/kerl/somacore-1.0.3)
Author: John Kerl <kerl.john.r@gmail.com>
Date:   Wed Jun 7 13:35:52 2023 -0400

    code-review feedback

commit d01fee32f2c107f95ffb2622effe37a46bf299cd
Author: John Kerl <kerl.john.r@gmail.com>
Date:   Wed Jun 7 11:27:33 2023 -0400

    code-review feedback

commit a7dc73340ceb11ebbaac6b6e012ddeb11c856db1
Author: John Kerl <kerl.john.r@gmail.com>
Date:   Wed Jun 7 11:17:37 2023 -0400

    somacore 1.0.3

commit 98aa63662af36d7b35d87841630741a79066de7c (origin/main, origin/HEAD, main)
Merge: 344d4eb 27d5ecc
Author: Seth Shelnutt <Shelnutt2@gmail.com>
Date:   Fri Mar 24 09:47:33 2023 -0400

    Merge pull request #1 from TileDB-Inc/ss/build-for-tiledb-channel

    Build for tiledb channel
```

```
$ git diff -r HEAD~1
diff --git a/recipe/meta.yaml b/recipe/meta.yaml
index 77196db..06d07ce 100644
--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -12,7 +12,7 @@ source:
 build:
   noarch: python
   script: {{ PYTHON }} -m pip install . -vv
-  number: 0
+  number: 1

 requirements:
   host:
@@ -23,14 +23,15 @@ requirements:
     - pip
   run:
     - python >=3.7
-    - anndata >=0.9.1
-    - attrs >=23.1
+    - anndata
+    - attrs >=22.1
     - numba
-    - numpy >=1.20
+    - numpy >=1.21
     - pandas
-    - pyarrow >=12.0
+    - pyarrow
     - scipy
-    - typing-extensions
+    - typing-extensions >=4.1 # For LiteralString
+# See also https://github.com/single-cell-data/SOMA/blob/VERSION-GOES-HERE/pyproject.toml

 test:
   imports:
```

Yet:

* After `git add` of the changes and `git commit` and `git push`
* At https://github.com/TileDB-Inc/somacore-feedstock/pull/2/commits I am still seeing only the _two_ commits, not the third -- after quadruple-checking everything locally & remotely, hard page-reload, etc etc etc

**Comment by johnkerl** [comment]
Screenshot 2023-06-07 at 1 41 43 PM

**Comment by johnkerl** [comment]
Screenshot 2023-06-07 at 1 42 34 PM

File: recipe/meta.yaml [hunk 2]
```diff
@@ -12,7 +12,7 @@ source:
 build:
   noarch: python
   script: {{ PYTHON }} -m pip install . -vv
-  number: 0
+  number: 1
```
Comment (jdblischak) [inline recipe/meta.yaml]: This should stay at `number: 0` like you had it originally

**Review by jdblischak — APPROVED** [review]

**Pull request merged**